File: samplejava/src/main/java/pyxis/uzuki/live/richutils/samplejava/InAppActivity.java
package pyxis.uzuki.live.richutils.samplejava;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.Button;

import java.util.ArrayList;

import pyxis.uzuki.live.richutilskt.module.iap.RInAppBilling;
import pyxis.uzuki.live.richutilskt.module.iap.Sku;

public class InAppActivity extends AppCompatActivity {
    private RInAppBilling utils;
    private String item1Sku = "item_1";
    private String item2Sku = "item_2";
    private String item3Sku = "item_3";

    private Button btn1;
    private Button btn2;
    private Button btn3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btn1 = (Button) findViewById(R.id.button);
        btn2 = (Button) findViewById(R.id.button2);
        btn3 = (Button) findViewById(R.id.button3);

        utils = new RInAppBilling(this, "");

        utils.setOnInAppBillingCallback((responseCode, transaction) -> {
            if (responseCode == RInAppBilling.PURCHASE_SUCCESS) {
                Log.d("Purchase", "Success!!");
                Log.d("Purchase", transaction.getPurchaseInfo());
                Log.d("Purchase", transaction.getDataSignature());
                Log.d("Purchase", transaction.getPurchaseToken());
            } else {
                Log.d("Purchase", "Failed!!");
            }
        });

        utils.setOnInAppConsumeCallback((responseCode, transaction) -> {
            if (responseCode == RInAppBilling.PURCHASE_SUCCESS) {
                Log.d("Consume", "Consume success!!");
                Log.d("Consume", transaction.getPurchaseInfo());
                Log.d("Consume", transaction.getDataSignature());
                Log.d("Consume", transaction.getPurchaseToken());
            } else {
                Log.d("Consume", "Failed!!");
            }
        });

        utils.bindInAppBilling();

        ArrayList<String> items = new ArrayList<>();
        items.add(item1Sku);
        items.add(item2Sku);
        items.add(item3Sku);

        ArrayList<Sku> availablePackage = utils.getAvailableInappPackage(items);
        for (Sku sku : availablePackage) {
            Log.d(sku.getProductId(), sku.getPrice() + ": " + sku.getTitle() + " : " + sku.getType());
        }

        btn1.setOnClickListener(v -> utils.purchase(item1Sku));

        btn2.setOnClickListener(v -> utils.purchase(item2Sku));

        btn3.setOnClickListener(v -> utils.purchase(item3Sku));
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        utils.unBindInAppBilling();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (utils != null) {
            utils.onActivityResult(requestCode, resultCode, data);
        }
        super.onActivityResult(requestCode, resultCode, data);
    }
}

File: sample/src/main/java/pyxis/uzuki/live/richutilssample/InAppActivity.kt
package pyxis.uzuki.live.richutilssample

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.Button
import kotlinx.android.synthetic.main.activity_inapp.*

import java.util.ArrayList
import pyxis.uzuki.live.richutilskt.module.iap.RInAppBilling
import pyxis.uzuki.live.richutilskt.module.iap.Sku
import pyxis.uzuki.live.richutilskt.module.iap.Transaction

class InAppActivity : AppCompatActivity() {
    private var utils: RInAppBilling? = null
    private val item1Sku = "item_1"
    private val item2Sku = "item_2"
    private val item3Sku = "item_3"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_inapp)

        utils = RInAppBilling(this, "")

        utils?.setOnInAppBillingCallback { responseCode, transaction ->
            if (responseCode == RInAppBilling.PURCHASE_SUCCESS) {
                Log.d("Purchase", "Success!!")
                Log.d("Purchase", transaction?.purchaseInfo)
                Log.d("Purchase", transaction?.dataSignature)
                Log.d("Purchase", transaction?.purchaseToken)
            } else {
                Log.d("Purchase", "Failed!!")
            }
        }

        utils?.setOnInAppConsumeCallback { responseCode, transaction ->
            if (responseCode == RInAppBilling.PURCHASE_SUCCESS) {
                Log.d("Consume", "Consume success!!")
                Log.d("Consume", transaction?.purchaseInfo)
                Log.d("Consume", transaction?.dataSignature)
                Log.d("Consume", transaction?.purchaseToken)
            } else {
                Log.d("Consume", "Failed!!")
            }
        }

        utils?.bindInAppBilling()

        val items = arrayListOf(item1Sku, item2Sku, item3Sku)
        val availablePackage = utils?.getAvailableInappPackage(items)

        if (availablePackage != null) {
            for ((productId, type, price, _, _, title) in availablePackage) {
                Log.d(productId, "$price: $title : $type")
            }
        }

        button.setOnClickListener { utils?.purchase(item1Sku) }
        button2.setOnClickListener { utils?.purchase(item2Sku) }
        button3.setOnClickListener { utils?.purchase(item3Sku) }
    }

    override fun onDestroy() {
        super.onDestroy()
        utils!!.unBindInAppBilling()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        if (utils != null) {
            utils!!.onActivityResult(requestCode, resultCode, data)
        }
        super.onActivityResult(requestCode, resultCode, data)
    }
}
